client/my-sites/plans-grid/components/test/actions.tsx

```diff
@@ -276,6 +276,73 @@ describe( 'PlanFeatures2023GridActions', () => {
 
 				expect( upgradeButton ).toHaveTextContent( 'Upgrade – $20' );
 			} );
+
+			test( 'should render the trial ended message when user is not eligible anymore and is on the new-hosted-site flow', () => {
```

Comment:
I don't think we need these granular tests inside `plans-grid`.

Once we make proper use of `actionOverrides`, you can migrate the tests into `plans-features-main` and ensure the right "override" is passed through. Then the only test we need under `plans-grid` is that "it renders the correct override when present"

Reply:
That makes sense, thanks for the suggestion, [updated it here](https://github.com/Automattic/wp-calypso/pull/86249/commits/53c31c0794519b5de88ea4eadbcb90df1121445b).